std/src/main/scala/cats/std/list.scala

```diff
@@ -42,8 +42,8 @@ trait ListInstances {
         Fold.partialIterate(fa)(f)
 
       def traverse[G[_], A, B](fa: List[A])(f: A => G[B])(implicit G: Applicative[G]): G[List[B]] = {
-        val gba = G.pure(ListBuffer.empty[B])
-        val gbb = fa.foldLeft(gba)((buf, a) => G.map2(buf, f(a))(_ += _))
+        val gba = G.pure(Vector.empty[B])
+        val gbb = fa.foldLeft(gba)((buf, a) => G.map2(buf, f(a))(_ :+ _))
```

Comment:
List failed the parallel composition law when it was using mutable state in traverse.
I didn't want to switch to an immutable list because we're appending.